src/services/infra/InfraService.ts

```diff
@@ -91,4 +104,123 @@ export default class InfraService {
       throw err
     }
   }
+
+  launchSite = async (
+    submissionId: string,
+    requestor: User,
+    agency: User,
+    repoName: string,
+    primaryDomain: string,
+    subDomainSettings: SubDomainSettings
+  ) => {
+    // call amplify to trigger site launch process
+    try {
+      // Set up domain association using LaunchesService
+      const {
+        appId,
+        siteId,
+      } = await this.launchesService.configureDomainInAmplify(
+        repoName,
+        primaryDomain,
+        subDomainSettings
+      )
+      logger.info(
+        `Created Domain association for ${repoName} to ${primaryDomain}`
+      )
+
+      // Get DNS records from Amplify
+      const dnsInfo = await this.launchesService.getDomainAssociationRecord(
+        primaryDomain,
+        appId
+      )
+
+      const certificationRecord = dnsInfo.domainAssociation?.certificateVerificationDNSRecord?.split(
```

Comment:
If `dnsInfo.domainAssociation?.certificateVerificationDNSRecord` is null, would that cause the `split()` to fail?